src/Parser.java

```diff
@@ -0,0 +1,114 @@
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+
+/**
+ * Parser class that deals with loading tasks from the file and
+ * saving tasks in the file.
+ */
+public class Parser {
+    /**
+     * Evaluates the syntax of the input command and executes it,
+     * throwing a DukeException if the syntax is correct.
+     * @param taskList the TaskList object associated with the current Duke object
+     * @param ui the Ui object associated with the current Duke object
+     * @param storage the Storage object associated with the current Duke object
+     */
+    public static void action(TaskList taskList, Ui ui, Storage storage) {
+        ui.welcome();
+        storage.load(taskList, ui);
+        ui.printSaved(taskList);
+        Scanner sc = new Scanner(System.in);
+        while (true) {
+            String input = sc.nextLine();
+            try {
+                if (input.equals("bye")) {
+                    ui.printBye();
+                    break;
+                } else if (input.equals("list")) {
+                    ui.printList(taskList);
+                } else if (input.startsWith("done")) {
+                    String[] number = input.split("done ");
+                    int num = Integer.parseInt(number[1]);
+                    Task task = taskList.getTask(num);
+                    task.markAsDone();
+                    storage.saveTasks(taskList, ui);
+                    ui.printDone(taskList.getTask(num));
+                } else if (input.startsWith("delete")) {
+                    String[] number = input.split("delete ");
+                    int num = Integer.parseInt(number[1]);
+                    Task task = taskList.getTask(num);
+                    taskList.deleteTask(num);
+                    ui.printDelete(taskList, task);
+                    storage.saveTasks(taskList, ui);
+                } else if (input.startsWith("todo")) {
+                    String[] array = input.split("todo ");
+                    if (array.length < 2) {
+                        throw new DukeException("The description of a todo cannot be empty!");
+                    } else {
+                        String des = array[1];
```

Comment:
I think this variable can be named as "description" to give the reader a better idea.